runtime/compiler/runtime/JITServerIProfiler.cpp

```diff
@@ -91,26 +88,25 @@ JITServerIProfiler::deserializeMethodEntry(TR_ContiguousIPMethodHashTableEntry *
    return entry;
    }
 
-TR_ContiguousIPMethodHashTableEntry
-TR_ContiguousIPMethodHashTableEntry::serialize(TR_IPMethodHashTableEntry *entry)
+void
+TR_ContiguousIPMethodHashTableEntry::serialize(TR_IPMethodHashTableEntry *entry, TR_ContiguousIPMethodHashTableEntry *serialEntry)
    {
-   TR_ContiguousIPMethodHashTableEntry serialEntry;
-   memset(&serialEntry, 0, sizeof(TR_ContiguousIPMethodHashTableEntry));
-   serialEntry._method = entry->_method;
-   serialEntry._otherBucket = entry->_otherBucket;
+   serialEntry->_method = entry->_method;
+   serialEntry->_otherBucket = entry->_otherBucket;
+   serialEntry->_callerCount = 0;
 
    size_t callerIdx = 0;
    for (TR_IPMethodData *caller = &entry->_caller; caller; caller = caller->next)
       {
       if (callerIdx >= TR_IPMethodHashTableEntry::MAX_IPMETHOD_CALLERS)
          break;
-      auto &serialCaller = serialEntry._callers[callerIdx];
+      auto &serialCaller = serialEntry->_callers[callerIdx];
       serialCaller._method = caller->getMethod();
       serialCaller._pcIndex = caller->getPCIndex();
       serialCaller._weight = caller->getWeight();
+      serialEntry->_callerCount++;
```

Comment:
Small tweak: take this increment out of the `for` loop and at the end just add `serialEntry->_callerCount=callerIdx;`